tests/yast2_gui/yast2_control_center.pm

```diff
@@ -123,220 +136,292 @@ sub run() {
 
     #   start Online Migration
     assert_and_click "yast2_control-center_online-migration";
-    assert_and_click "yast2_control-center-online-migration_cancel";
+    assert_screen "yast2_control-center-online-migration_cancel", 60;
+    send_key "alt-c";
     assert_screen 'yast2-control-center-ui';
 
     #   start Partitioner
     assert_and_click "yast2_control-center_partitioner";
-    assert_screen "yast2_control-center-partitioner_warning";
+    assert_screen "yast2_control-center-partitioner_warning", 60;
     assert_and_click "yast2_control-center-partitioner_abort";
     assert_screen 'yast2-control-center-ui';
 
     #   start Service Manager
     assert_and_click "yast2_control-center_service-manager";
-    assert_screen "yast2_control-center-service-manager";
+    assert_screen "yast2_control-center-service-manager_list", 60;
     assert_and_click "yast2_control-center-service-manager_ok";
     assert_screen 'yast2-control-center-ui';
 
     #   start Authentication Server
     #	need to scroll down to get other modules
     send_key "down";
     assert_and_click "yast2_control-center_authentication-server";
-    assert_screen "yast2_control-center-authentication-server_install";
-    assert_and_click "yast2_control-center-authentication-server_install-cancel";
-    assert_and_click "yast2_control-center-authentication-server_install-error";
-    assert_screen 'yast2-control-center-ui';
+    if (check_screen "yast2_control-center-authentication-server_install") {
```

Comment:
can you create a needle for both cases "if" and "else", check on both with `assert_screen` and use `match_has_tag`?